File: python/kindergarten-garden/kindergarten_garden.py
class Garden:
    def __init__(
        self,
        diagram,
        students=[
            "Alice",
            "Bob",
            "Charlie",
            "David",
            "Eve",
            "Fred",
            "Ginny",
            "Harriet",
            "Ileana",
            "Joseph",
            "Kincaid",
            "Larry",
        ],
    ):
        self.diagram = diagram.split("\n")
        self.students = sorted(students)
        self.num_students = len(self.students)

    def get_plant(self, plant):
        match plant:
            case "V":
                return "Violets"
            case "R":
                return "Radishes"
            case "C":
                return "Clover"
            case "G":
                return "Grass"
            case _:
                return None

    def plants(self, student):
        index = self.students.index(student)
        plants = []
        for i in range(2):
            for j in range(2):
                plants.append(self.get_plant(self.diagram[i][index * 2 + j]))
        return plants

File: java/kindergarten-garden/src/main/java/KindergartenGarden.java
import java.util.List;
import java.util.ArrayList;

class KindergartenGarden {
    private String gardenRow1;
    private String gardenRow2;
    
    KindergartenGarden(String garden) {
        String[] g = garden.split("\n");
        this.gardenRow1 = g[0];
        this.gardenRow2 = g[1];
    }

    List<Plant> getPlantsOfStudent(String student) {
        int place = (student.charAt(0) - 'A') * 2;
        List<Plant> plants = new ArrayList<>();
        plants.add(Plant.getPlant(gardenRow1.charAt(place)));
        plants.add(Plant.getPlant(gardenRow1.charAt(place + 1)));
        plants.add(Plant.getPlant(gardenRow2.charAt(place)));
        plants.add(Plant.getPlant(gardenRow2.charAt(place + 1)));
        return plants;
    }

}
